File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--add-feature-event-receivers_2.vb
    Public Overrides Sub FeatureDeactivating(ByVal properties As SPFeatureReceiverProperties)
        Try
            ' Get reference to SharePoint site.
            Dim site As SPSite = New SPSite("http://localhost")
            Dim web As SPWeb = site.OpenWeb("/")
            ' Get reference to Announcements list.
            Dim announcementsList As SPList = web.Lists("Announcements")
            ' Add new announcement to Announcements list.
            Dim oListItem As SPListItem = announcementsList.Items.Add
            oListItem("Title") = ("Deactivated Feature: " + properties.Definition.DisplayName)
            oListItem("Body") = (properties.Definition.DisplayName + (" was deactivated on: " + DateTime.Now.ToString))
            oListItem.Update()
        Catch e As Exception
            Console.WriteLine(("Error: " + e.ToString))
        End Try
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--add-feature-event-receivers_2.cs
        public override void FeatureDeactivating(SPFeatureReceiverProperties properties)
        {
            try
            {
                // Get reference to SharePoint site.
                SPSite site = new SPSite("http://localhost");
                SPWeb web = site.OpenWeb("/");
                // Get reference to Announcements list.
                SPList announcementsList = web.Lists["Announcements"];

                // Add new announcement to Announcements list.
                SPListItem oListItem = announcementsList.Items.Add();
                oListItem["Title"] = "Deactivated Feature: " + properties.Definition.DisplayName;
                oListItem["Body"] = properties.Definition.DisplayName + " was deactivated on: " + DateTime.Now.ToString();
                oListItem.Update();

            }

            catch (Exception e)
            {
                Console.WriteLine("Error: " + e.ToString());
            }
            
        }